core/src/test/java/org/gbif/pipelines/core/parsers/multimedia/MultimediaParserTest.java

```diff
@@ -0,0 +1,265 @@
+    assertEquals("Geranium Plume Moth 0032 description", multimedia.getDescription());
+    assertEquals("BY-NC-SA 2.0", multimedia.getLicense());
+    assertEquals("Moayed Bahajjaj", multimedia.getCreator());
+    assertEquals("2012-03-29", multimedia.getCreated().toString());
+    assertEquals("http://www.flickr.com/photos/70939559@N02/7039524065", multimedia.getReferences().toString());
+    assertEquals("http://farm8.staticflickr.com/7093/7039524065_3ed0382368.jpg", multimedia.getIdentifier().toString());
+  }
+
+  @Test
+  public void testInterpretAudubonExtension() {
+
+    Map<String, String> audubonExtension = new ExtendedRecordCustomBuilder.MultimediaExtensionBuilder().accessURI(
+      "http://specify-attachments-saiab.saiab.ac.za/originals/sp6-3853933608872243693.att.JPG")
+      .identifier("d79633d3-0967-40fa-9557-d6915e4d1353")
+      .format("jpg")
+      .title("Geranium Plume Moth 0032")
+      .description("Geranium Plume Moth 0032 description")
+      .derivedFrom("http://farm8.staticflickr.com/7093/7039524065_3ed0382368.jpg")
+      .license("BY-NC-SA 2.0")
+      .creator("Moayed Bahajjaj")
+      .created("2012-03-29")
+      .build();
+
+    Map<String, String> audubonExtension2 = new ExtendedRecordCustomBuilder.MultimediaExtensionBuilder().identifier(
+      "http://www.flickr.com/photos/70939559@N02/7039524065.jpg").build();
+
+    ExtendedRecord extendedRecord = new ExtendedRecordCustomBuilder().id("123")
+      .addExtensionRecord(Extension.AUDUBON, audubonExtension)
+      .addExtensionRecord(Extension.AUDUBON, audubonExtension2)
+      .build();
+
+    ParsedField<List<ParsedMultimedia>> parsedMultimedia = MultimediaParser.parseMultimedia(extendedRecord);
+
+    assertEquals(2, parsedMultimedia.getResult().size());
+    for (ParsedMultimedia media : parsedMultimedia.getResult()) {
+      assertEquals(MediaType.StillImage, media.getType());
+      assertNotNull(media.getIdentifier());
+    }
+
+    ParsedMultimedia multimedia = parsedMultimedia.getResult().get(1);
+    assertEquals("image/jpeg", multimedia.getFormat());
+    assertEquals("Geranium Plume Moth 0032", multimedia.getTitle());
+    assertEquals("Geranium Plume Moth 0032 description", multimedia.getDescription());
+    assertEquals("BY-NC-SA 2.0", multimedia.getLicense());
+    assertEquals("Moayed Bahajjaj", multimedia.getCreator());
+    assertEquals("2012-03-29", multimedia.getCreated().toString());
+    assertEquals("http://specify-attachments-saiab.saiab.ac.za/originals/sp6-3853933608872243693.att.JPG",
+                 multimedia.getIdentifier().toString());
+  }
+
+  /**
+   * If the information about the same image (same URI) is given in the core AND the extension we should use the one
+   * from the extension (richer data).
+   */
+  @Test
+  public void testExtensionsPriority() {
+    Map<String, String> audubonExtension = new ExtendedRecordCustomBuilder.MultimediaExtensionBuilder().accessURI(
```

Comment:
For it looks a bit weird, I lost context why it is a map, not a MultimediaExtension class, maybe would be better create a class - MultimediaExtension class

Reply:
It is because in the ExtendedRecord the extensions are a map and every extension has a list of maps (as I understood it's because one record can have several extension rows associated to it):

```
java.util.Map<java.lang.String,java.util.List<java.util.Map<java.lang.String,java.lang.String>>> extensions;
```

and here I'm just creating a ExtendedRecord with extensions for testing. This map is like the core terms map.